File: videojuegoDeChinpokomon/src/Ataque/GarraMecanica.java
package Ataque;

import Chimpokomon.Chimpokomon;
import logger.Logger;

public class GarraMecanica extends Ataque {
    public static Logger logger = Logger.getInstance();
    public GarraMecanica(Integer dañoValor, String nombre) {
        super(dañoValor, nombre);
        dañoValor = 2;
        nombre = "GarraMecanica";
    }

    @Override
    public void dañar(Chimpokomon chipoAtacante, Chimpokomon chipoEnemigo) {
        double random = Math.random();
        if (chipoAtacante.tieneVentajaSobre(chipoEnemigo)) {
            if (random < 0.5) {
                chipoEnemigo.restarVida((int) (chipoEnemigo.getNivelDeVida() * 0.5));
                logger.info(chipoAtacante.getNombre() + " Realizo un ataque critico");
            } else {
                chipoEnemigo.restarVida(this.getdañoValor() + 2);
            }
        } else {
            if (random < 0.1) {
                chipoEnemigo.restarVida((int) (chipoEnemigo.getNivelDeVida() * 0.5));
                logger.info(chipoAtacante.getNombre() + " Realizo un ataque critico");
            } else {
                chipoEnemigo.restarVida(this.getdañoValor());
            }
        }

    }
}

File: ChimpokomonPython/ModuloSRC/Ataque/GarraMecanica.py
from Ataque.Ataque import Ataque
import random 

class GarraMecanica(Ataque):
    def __init__(self):
            Ataque.__init__(self, "Garra Mecanica", 2, 2)

    def dañar(self,  chinpokomon, poder):
        ram = random.randint(1, 10)
    
        if ram == 1:
            chinpokomon.restarVida(chinpokomon.getNivelDeVida() * 0.5)
        else:        
            chinpokomon.restarVida(poder)
      
